managers/BaseManager.js

```diff
@@ -43,12 +43,17 @@ class BaseManager {
     logger.info('Trying to acquire processing lock for request with options: ', changedOptions);
     // Set lockedManager annotations to true
     const patchBody = _.cloneDeep(changeObjectBody);
-    let currentAnnotations = patchBody.metadata.annotations;
-    let patchAnnotations = currentAnnotations ? currentAnnotations : {};
+    const metadata = patchBody.metadata;
+    const patchAnnotations = metadata.annotations ? metadata.annotations : {};
     patchAnnotations.lockedByManager = config.broker_ip;
-    patchBody.metadata.annotations = patchAnnotations;
+    metadata.annotations = patchAnnotations;
     const resourceDetails = eventmesh.apiServerClient.parseResourceDetailsFromSelfLink(changeObjectBody.metadata.selfLink);
-    return eventmesh.apiServerClient.updateResource(resourceDetails.resourceGroup, resourceDetails.resourceType, changeObjectBody.metadata.name, patchBody)
+    return eventmesh.apiServerClient.updateResource({
```

Comment:
how are we ensuring we dont have two simultaneous updates and both passing?

Reply:
metadata contains a field called `resourceVersion ` number, and update can only go through if the current resourceVersion is same as one being passed in metadata.
Hence simultaneous updates can't go through as first update will increase the resourceVersion hence second update fails